src/api/app/controllers/webui/request_controller.rb

```diff
@@ -503,7 +503,9 @@ def staging_status(request, target_project)
   end
 
   def cache_diff_data
-    return unless (@refresh = @action[:diff_not_cached])
+    return @diff_not_cached = true unless defined?(@action)
+
+    return unless (@diff_not_cached = @action[:diff_not_cached])
```

Comment:
I guess that checking for `defined?(@action)` is now needed because of the previous commit, which moves the assignment to the `@refresh` variable into the `cache_diff_data before_action` method.

Currently, without introducing the `cache_diff_data before_action` method, the variable `@action` gets assigned before `@refresh` gets assigned, [here](https://github.com/openSUSE/open-build-service/blob/8f5ab39aea4552297a35bd9fa4e4882625aea43a/src/api/app/controllers/webui/request_controller.rb#L159-L163), for example.

In these cases, shouldn't we keep assigning the `@refresh` variable after `@action` gets assigned?